katas/content/multi_qubit_gates/toffoli_gate/Verification.qs

```diff
@@ -0,0 +1,58 @@
+namespace Kata.Verification {
+    open Microsoft.Quantum.Katas;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Convert;
+
+    operation PrepareState(angle:Double,qs : Qubit[]) : Unit is Adj + Ctl {
+        for i in 0 .. Length(qs)-1{
+            Ry(angle,qs[i]);
+        }
+    }
+
+    operation  ToffoliGate (qs : Qubit[]) : Unit is Adj + Ctl {
+        CCNOT(qs[0], qs[1], qs[2]);
+    }
+
+    operation CheckOperationsEquivalenceOnInitialStateStrict(
+        initialState : Qubit[] => Unit is Adj,
+        op : (Qubit[] => Unit is Adj + Ctl),
+        reference : (Qubit[] => Unit is Adj + Ctl),
+        inputSize : Int
+    ) : Bool {
+        use (control, target) = (Qubit(), Qubit[inputSize]);
+        within {
+            H(control);
+            initialState(target);
+        }
+        apply {
+            Controlled op([control], target);
+            Adjoint Controlled reference([control], target);
+        }
+        let isCorrect = CheckAllZero([control] + target);
+        ResetAll([control] + target);
+        isCorrect
+    }
+    
+    operation CheckSolution() : Bool {
```

Comment:
This task (as well as 2.4 and 2.6) should be tested as unitaries comparison indeed, since they accept generic input states, not specific ones (see [my comment](https://github.com/microsoft/qsharp/pull/1389#pullrequestreview-2013016417)).